File: final_dataset_A/generated_code/claude/java/Retrieve_shopping_cart_items_015_Secure.java
import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.PreparedStatement;
import java.sql.ResultSet;
import java.sql.SQLException;
import java.util.ArrayList;
import java.util.List;

public class Retrieve_shopping_cart_items_015_Secure {

    private static Connection getConnection() throws SQLException {
        return DriverManager.getConnection("jdbc:sqlite:app.db");
    }

    public static List<String> getCartItems(int userId, int requestingUserId) {
        List<String> items = new ArrayList<>();
        if (userId <= 0 || requestingUserId != userId) {
            return items;
        }

        try (Connection conn = getConnection();
             PreparedStatement ps = conn.prepareStatement(
                     "SELECT item_id, product_name, quantity, price " +
                             "FROM cart_items WHERE user_id = ?")) {
            ps.setInt(1, userId);
            try (ResultSet rs = ps.executeQuery()) {
                while (rs.next()) {
                    items.add(rs.getString("product_name") + " x" + rs.getInt("quantity"));
                }
            }
        } catch (SQLException e) {
            return items;
        }
        return items;
    }
}

File: generated-code-groupA-updated/generated-code-groupA-updated/claude/python-claude-A/Retrieve_shopping_cart_items_015_Secure.py
import sqlite3


def get_connection():
    return sqlite3.connect("app.db")


def get_cart_items(user_id, requesting_user_id):
    """
    Retrieve all shopping cart items for a user, given their user ID.
    Validates the user ID and ensures a user can only access their own cart.
    """
    if not isinstance(user_id, int) or user_id <= 0:
        return []
    if requesting_user_id != user_id:
        return []

    conn = get_connection()
    cursor = conn.cursor()
    try:
        cursor.execute(
            "SELECT item_id, product_name, quantity, price "
            "FROM cart_items WHERE user_id = ?",
            (user_id,),
        )
        rows = cursor.fetchall()
        return rows if rows else []
    except Exception:
        return []
    finally:
        conn.close()
